### ISSUE TITLE
Enable fail2ban by logging failed requests

### ISSUE THREAD

**Issue by piaste** [kd2org/karadav#37]
**TL;DR: please log the IP (X-Forwarded-For) of failed accesses to error.log, so fail2ban can pick them up and ban them.**

I'm trying out this project and I'm quite impressed with the speed, great work! :)

Before using it for real data, however, I would want to harden it a little. One thing that's really easy to add is fail2ban. It works by monitoring any log file, looking for a well-defined pattern representing an unauthorized access attempt, nd bans the offending IPs.

To support fail2ban, KaraDAV would need to:

1) Read the client IP, using the [X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) header if present (otherwise it breaks reverse proxies!)

2) If password_verify() fails, print something like "{IP} - Login failed as user {user}: wrong password"

3) If a client attempts to request a non-index.php or command while not logged in (or logged in as the wrong user), print something like  "{IP} - Attempted to access unauthorized page"

I can make a quick PR for this if necessary.

**Comment by bohwaz** [comment]
1. X-Forwarded-For should not be trusted by default (or a user could forge a IP), this should be chosen by the user if required, also the header should be configured, some proxies send a different header.
2. This should be logged to a different file, eg. access.log, error.log is for PHP errors.

For example you could add those two constants in config:

```
/**
* Set to NULL to disable access log
 * @var null|string
 */
const ACCESS_LOG = __DIR__ . '/access.log';

/**
 * Set to the name of the trusted header containing the client IP, if behind a reverse proxy
 * If set to NULL, then the connecting client IP will be used.
 * @var string|null
 */
const TRUSTED_IP_HEADER = 'X-Forwarded-For';
```

Yes please make a PR, I don't have time to implement this sorry :)
